upgrade.sh

```diff
@@ -52,6 +52,66 @@ function unbound_psp_check() {
     echo "cray-unbound-coredns-psp check Done"
 }
 
+# Function to check cray-sysmgmt-health chart with app version 9.3.1 for prometheus-operator and retain old PVs data.
+function sysmgmt_health() {
+    echo "Checking for chart version of cray-sysmgmt-health"
+    version="45.1"
+    if [ $(helm ls -o json --namespace sysmgmt-health| jq -r --argjson version $version '.[] | select(.app_version | sub(".[0-9]$";"") | tonumber | . < $version).name') ]
+    then
+    prom_pvc="prometheus-cray-sysmgmt-health-promet-prometheus-db-prometheus-cray-sysmgmt-health-promet-prometheus-0"
+    alert_pvc="alertmanager-cray-sysmgmt-health-promet-alertmanager-db-alertmanager-cray-sysmgmt-health-promet-alertmanager-0"
+    echo "Get PV for both prometheus and Alertmanager"
+    prom_pv=$(kubectl get pvc -n sysmgmt-health --output=json $prom_pvc |jq '.spec."volumeName"')
+    alert_pv=$(kubectl get pvc -n sysmgmt-health --output=json $alert_pvc |jq '.spec."volumeName"')
+    prom_pv="${prom_pv//[\",]}"
+    alert_pv="${alert_pv//[\",]}"
+    echo "$prom_pv"
+    echo "$alert_pv"
+
+    # Patch the PersistenceVolume created/used by the prometheus-operator and alertmanager to Retain claim policy
+    kubectl patch pv/$prom_pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+    kubectl patch pv/$alert_pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}'
+
+    # Uninstall the cray-sysmgmt-health release
+    helm ls -o json --namespace sysmgmt-health| jq -r --argjson version $version '.[] | select(.app_version | sub(".[0-9]$";"") | tonumber | . < $version).name'| xargs -L1 helm uninstall --namespace sysmgmt-health
+    
+    # Delete the existing PersistentVolumeClaim, and verify PV become Released
+    kubectl delete pvc/$prom_pvc -n sysmgmt-health
+    kubectl delete pvc/$alert_pvc -n sysmgmt-health
+    alert_pv_phase=$(kubectl get pv $prom_pv -o json |jq '.status."phase"')
+    prom_pv_phase=$(kubectl get pv $alert_pv -o json |jq '.status."phase"')
+    prom_pv_phase="${prom_pv_phase//[\",]}"
+    alert_pv_phase="${alert_pv_phase//[\",]}"
+    echo "Verifying whether PVs became Released or not."
+    if [ "$alert_pv_phase" == Released ] && [ "$prom_pv_phase" == Released ]; then
+        echo "Both Prometheus and Alertmanager PVs are Released"
+    else
+        echo >&2 "PVs are not Released. Verify if PV exists or not."
```

Comment:
Should probably exit here if the pv's need to be released at this point instead of continuing. Then if the issue was ephemeral this function can be reran. Other option is to loop a certain amount of times on the condition before failing as it stands if the pv isn't released really fast this script will always fail at this point if there is a delay in the pv becoming Released.

Reply:
Added exit if the condition fails.
The PV becomes released within a fraction of seconds, but added sleep time of 5 seconds to be on safer side.
Also added pv names in the error cases so that people know what to look for.